### ISSUE TITLE
Bank Account - Navigation Problem in "Get assistance" page after a reload

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#31892]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.3-6
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:** @

## Action Performed:
1. Go to settings > Workspace > choose a works space(currency should be USD)
2. Bank account > Connect Manually
3. Click on the Question mark at the top.
4. Reload the Get Assistance page
5. Click on the back button.

## Expected Result:
page navigates to the "Connect bank account" page

## Actual Result:
page navigates to the "Workspace" page

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/c62fa74d-72cb-4743-8c72-f9cda78cfb2c

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
Production - different behavior.

https://github.com/Expensify/App/assets/115492554/1e139c60-e42f-45ae-ae81-d68ac7dc9a6d

**Comment by abzokhattab** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Back from assistance page after reload doesnt go back to the connect bank page
### What is the root cause of that problem?

the back navigation is set to the workspace settings url: 

https://github.com/Expensify/App/blob/d521dd6783c382297e06a429701ae1936f808041/src/pages/GetAssistancePage.js#L85

### What changes do you think we should make in order to solve the problem?

since the url of the get assistance page doesnt contain the workspace id 

https://staging.new.expensify.com/get-assistance/WorkspaceBankAccount

we need to add `backTo` param when clicking on the get assistance url to the save the prev route and go back to it on clicking the back button

**Comment by it-education-md** [comment]
I think https://github.com/Expensify/App/issues/31895#issuecomment-1827016675 is right

**Comment by mountiny** [comment]
I dont think this is a deploy blocker, fwiw it seems to work better than [production](https://github.com/Expensify/App/issues/31892#issuecomment-1826929184) 😂 

Demoting

**Comment by abzokhattab** [comment]
Hi @tienifr , I disagree with you the main idea was already mentioned in the first version (which is the same as yours). in the second version, i added the implementation steps for the solution proposed.

There is no breaking change in the proposal, both versions have the same RCA and proposed solution but the second one i added the code changes. 

<img width="624" alt="image" src="https://github.com/Expensify/App/assets/59809993/60ba3b26-698b-4328-9bbf-4cda285b4a18">

**Comment by allroundexperts** [comment]
Taking this one over!

**Comment by slafortune** [comment]
@allroundexperts do I need to assign you and unassign @ntdiary ?

**Comment by allroundexperts** [comment]
@slafortune Please do so.

**Comment by allroundexperts** [comment]
Thanks for the proposals everyone!

@abzokhattab Your initial proposal was too generic and lacked any details on how you were to implement the persistence of previous route. While concrete implementation details are not necessarily required for a proposal to be accepted, a general high level explanation of the `how` and `why` is a must.

@tienifr's [proposal](https://github.com/Expensify/App/issues/31892#issuecomment-1827143308) explains the root cause and proposes a clear solution as well. I think we should go with them.

🎀 👀 🎀 C+ reviewed

**Comment by cead22** [comment]
@abzokhattab thanks for the note, I definitely understand where you're coming from, since we try to balance succinctness with clarity, and I think in this case a little more detail would've made your proposal more immediately obvious as to what it was doing.

I hope this doesn't discourage you from making other proposals, but in order to move this forward, I'm going to assign @tienifr to this issue

**Comment by slafortune** [comment]
I am so sorry for the delay with this @tienifr !  I hadn't noticed this once it moved to monthly and didn't have the payment date in the title 😭 


Paid!

**Comment by slafortune** [comment]
@allroundexperts Please request via NewDot manual requests for the Reviewer role ($500)

**Comment by JmillsExpensify** [comment]
$500 approved for @allroundexperts based on comment above.
